src/libraries/System.Text.RegularExpressions/gen/System.Text.RegularExpressions.Generator.csproj

```diff
@@ -1,14 +1,18 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <!--
+      Source generators must target netstandard2.0, which doesn't support nullable reference types. In order
+      to enable the nullable reference type compiler checks, we also target NetCoreAppToolCurrent. Note that
+      this doesn't use the live shared framework but an LKG targeting pack instead to avoid layering concerns.
+    -->
+    <TargetFrameworks>netstandard2.0;$(NetCoreAppToolCurrent)</TargetFrameworks>
```

Comment:
@ViktorHofer I assume we don't need anything special in order for the generator that gets placed in the refpack be the netstandard2.0 one (as opposed to the NCA one), right?

Reply:
Picking the right asset for packages and the targeting pack is handled by the repo infrastructure. As documented in https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/project-guidelines.md#gen, P2P asset selection of multi-targeting source generator projects must be handled by the referencee. If explicit asset selection isn't done, Roslyn will fail to load the assembly:

```
"CSC : error CS8032: An instance of analyzer System.Text.RegularExpressions.Generator.RegexGenerator cannot be created from D:\repos\runtime\artifacts\bin\System.Text.RegularExpressions.Generator\Debug\net7.0\System.Text.RegularExpressions.Generator.dll : Could not load
file or assembly 'System.Runtime, Version=7.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.. [D:\repos\runtime\src\libraries\System.Private.Xml\src\System.Private.Xml.csproj::TargetFramework=net8.0-windows]"
```

In https://github.com/dotnet/runtime/pull/79613#discussion_r1048732169, @ericstj and I agreed on that asset selection should be the referencee's responsibility.

Reply:
Got it, my question was mostly if there was some change needed in order to force the netstandard 2.0 asset to be selected for the refpack (as I didn't see any related changes on the PR), but now I've found why that is not needed:
 https://github.com/dotnet/runtime/blob/2975fa4ca577b46d651f9380fda8cc3b39f1550b/src/libraries/Directory.Build.targets#L103

Reply:
@joperezr for reference see https://github.com/dotnet/runtime/blob/2975fa4ca577b46d651f9380fda8cc3b39f1550b/src/libraries/Directory.Build.targets#L100-L103.

Reply:
Jinx ;)